app/gui/view/graph-editor/src/component/node.rs

```diff
@@ -923,3 +926,36 @@ impl display::Object for Node {
         &self.model.display_object
     }
 }
+
+
+// === Positioning ===
+
+fn x_offset_to_node_center(node_width: f32) -> f32 {
+    node_width / 2.0
+}
+
+/// Calculate a position where to render the [`visualization::Container`] of a node, relative to
+/// the node's origin.
+fn visualization_offset(node_width: f32) -> Vector2 {
+    Vector2(x_offset_to_node_center(node_width), VISUALIZATION_OFFSET_Y)
+}
+
+fn bounding_box(
+    node_position: Vector2,
+    node_size: Vector2,
+    visualization_enabled_and_visible: bool,
+    visualization_size: Vector2,
+) -> BoundingBox {
+    let x_offset_to_node_center = x_offset_to_node_center(node_size.x);
+    let node_bbox_pos = node_position + Vector2(x_offset_to_node_center, 0.0) - node_size / 2.0;
+    let mut node_bbox = BoundingBox::from_position_and_size(node_bbox_pos, node_size);
+    if visualization_enabled_and_visible {
+        let visualization_offset = visualization_offset(node_size.x);
+        let visualization_pos = node_position + visualization_offset;
+        let visualization_bbox_pos = visualization_pos - visualization_size / 2.0;
+        let visualization_bbox =
+            BoundingBox::from_position_and_size(visualization_bbox_pos, visualization_size);
+        node_bbox.concat_mut(visualization_bbox);
+    }
```

Comment:
boundingBox is Copy, so it's fast to copy/clone it. It is better not to use mutable variables in the code, because then the code is easier to read and understand. In this code, you can use `concat_mut_ref` instead and not make `node_bbox` a mut var.

Reply:
Nice, thanks! ❤️ I was wishing I could make the code not use a mutable variable, but I didn't manage to connect all the dots to make it happen. Thanks for the great hint!!